libs/hdf-converters/src/conveyor-mapper.ts

```diff
@@ -0,0 +1,244 @@
+  } else {
+    desc = JSON.stringify(data);
+  }
+  return {
+    status: determineStatus(score),
+    code_desc: desc,
+    start_time: date,
+    run_time:  new Date(endTime).valueOf() - new Date(date).valueOf()
+  };
+}
+
+function childrenfinder(currLevel: Record<string, unknown>): string[][] {
+  const arr: string[][] = [];
+  for (const [sha, file] of Object.entries(currLevel)) {
+    if (_.has(file, 'name')) {
+      const name: string = _.get(file, 'name[0]') || '';
+      arr.push([sha, name]);
+    }
+    if (_.has(file, 'children')) {
+      const nextLevel: Record<string, unknown> = _.get(file, 'children') || {};
+      for ( const element of childrenfinder(nextLevel)) {
+        arr.push(element);
+      };
+    }
+  };
+  return arr;
+}
+function sha2filenameMapper(
+  results: Record<string, unknown>
+): Record<string, unknown> {
+  const toplevel = _.get(results, 'api_response.file_tree') as Record<
+    string,
+    unknown
+  >;
+  const shaMappings = {};
+  const arr = childrenfinder(toplevel)
+  arr.forEach((value) => {
+    const sha: string = value[0] || '';
+    const filename: string = value[1] || '';
+    _.set(shaMappings, sha, filename);
+  });
+  return shaMappings;
+}
+function determineStatus(score: number): ExecJSON.ControlResultStatus {
+  if (score == 0) {
+    return ExecJSON.ControlResultStatus.Passed;
+  }
+  return ExecJSON.ControlResultStatus.Failed;
+}
+
+function arrayifyObject(
+  parsed: Record<string, unknown>,
+  mappings: Record<string, unknown>
+): Record<string, unknown> {
+  const results = _.get(parsed, 'api_response.results') as Record<string, unknown>;
+  const newout: Record<string, unknown>[] = [];
+  _.forEach(results, (result) => {
+    const temp = result as Record<string, unknown>;
+    const sha: string = _.get(result, 'sha256') || '';
+    _.set(temp, 'filename', _.get(mappings, sha));
+    const descriptions = _.map(
+      _.get(temp, 'result.sections') as Record<string, unknown>[],
+      (val) =>
+        createDescription(
+          val as Record<string, unknown>,
+          _.get(temp, 'result.score') as number,
+          _.get(temp, 'response.milestones.service_started') as string,
+          _.get(temp, 'response.service_name') as string,
+          _.get(temp, 'response.milestones.service_completed') as string
+        )
+    );
+    if (descriptions.length == 0) {
+      descriptions.push({
+        status: ExecJSON.ControlResultStatus.Passed,
+        code_desc: 'NA',
+        start_time: _.get(temp, 'response.milestones.service_started') as string
+      });
+    }
+    _.set(temp, 'result.sections', descriptions);
+    newout.push(temp);
+  });
+  const groups = _.groupBy(newout, (result) => {
+    return _.get(result, 'response.service_name');
+  });
+  return groups;
+}
+
+function controlMappingConveyor(): MappedTransform<
+  ExecJSON.Control & ILookupPath,
+  ILookupPath
+> {
+  return {
+    id: {path: 'sha256'},
+    title: {path: 'filename'},
+    desc: '',
```

Comment:
do none of the results ever have a value that could serve as an overarching description for the control?